File: swagger/models/class.py
from .base_model import BaseModel

from .visit import Visit
from .client import Client
from .location import Location
from .resource import Resource
from .class_description import ClassDescription
from .staff import Staff
from .booking_window import BookingWindow

class Class(BaseModel):
	"""
	Represents a single class instance. The class meets at the start time,
	goes until the end time.

	Attributes:

		class_schedule_id
		The ID used to retrieve the class schedule for the desired class.

		visits
		Contains information about visits.

		clients
		Contains information about clients.

		location
		Contains information about a location.

		resource
		Contains information about a resource, such as a room, assigned to a
		class.

		max_capacity
		The maximum number of clients allowed in the class.

		web_capacity
		The maximum number of clients allowed to sign up online for the class.

		total_booked
		The total number of clients booked in the class.

		total_booked_waitlist
		The total number of booked clients on the waiting list for the class.

		web_booked
		The total number of clients who signed up online for the class.

		semester_id
		The ID of the semester that the class is a part of, if any.

		is_canceled
		When `true`, indicates that the class has been cancelled.<br />  When
		`false`, indicates that the class has not been cancelled.

		substitute
		When `true`, indicates that the class is being taught by a substitute
		teacher.<br />  When `false`, indicates that the class is being taught
		by its regular teacher.

		active
		When `true`, indicates that the class is shown to clients when in
		consumer mode.<br />  When `false`, indicates that the class is not
		shown to clients when in consumer mode.

		is_waitlist_available
		When `true`, indicates that the clients can be placed on a waiting
		list for the class.<br />  When `false`, indicates that the clients
		cannot be placed on a waiting list for the class.

		is_enrolled
		When `true`, indicates that the client with the given `ClientId` is
		enrolled in this class.<br />  When `false`, indicates that the client
		with the given `ClientId` is not enrolled in this class.

		hide_cancel
		When `true`, indicates that this class is hidden when cancelled.<br />
		When `false`, indicates that this class is not hidden when cancelled.

		id
		The unique identifier for the class.

		is_available
		When `true`, indicates that the client with the given client ID can
		book this class.<br />  When `false`, indicates that the client with
		the given client ID cannot book this class.

		start_date_time
		The time this class is scheduled to start.

		end_date_time
		The time this class is scheduled to end.

		last_modified_date_time
		The last time this class was modified.

		class_description
		Contains information that defines the class.

		staff
		Contains information about the staff member.

		booking_window
		Contains information about the window for booking.

		booking_status
		Contains the booking’s payment status.
	"""

	input_map = {
		'ClassScheduleId': 'class_schedule_id',
		'Visits': ['visits', Visit],
		'Clients': ['clients', Client],
		'Location': ('location', Location),
		'Resource': ('resource', Resource),
		'MaxCapacity': 'max_capacity',
		'WebCapacity': 'web_capacity',
		'TotalBooked': 'total_booked',
		'TotalBookedWaitlist': 'total_booked_waitlist',
		'WebBooked': 'web_booked',
		'SemesterId': 'semester_id',
		'IsCanceled': 'is_canceled',
		'Substitute': 'substitute',
		'Active': 'active',
		'IsWaitlistAvailable': 'is_waitlist_available',
		'IsEnrolled': 'is_enrolled',
		'HideCancel': 'hide_cancel',
		'Id': 'id',
		'IsAvailable': 'is_available',
		'StartDateTime': 'start_date_time',
		'EndDateTime': 'end_date_time',
		'LastModifiedDateTime': 'last_modified_date_time',
		'ClassDescription': ('class_description', ClassDescription),
		'Staff': ('staff', Staff),
		'BookingWindow': ('booking_window', BookingWindow),
		'BookingStatus': 'booking_status',
		}

	output_map = {
		'class_schedule_id': 'ClassScheduleId',
		'visits': 'Visits',
		'clients': 'Clients',
		'location': 'Location',
		'resource': 'Resource',
		'max_capacity': 'MaxCapacity',
		'web_capacity': 'WebCapacity',
		'total_booked': 'TotalBooked',
		'total_booked_waitlist': 'TotalBookedWaitlist',
		'web_booked': 'WebBooked',
		'semester_id': 'SemesterId',
		'is_canceled': 'IsCanceled',
		'substitute': 'Substitute',
		'active': 'Active',
		'is_waitlist_available': 'IsWaitlistAvailable',
		'is_enrolled': 'IsEnrolled',
		'hide_cancel': 'HideCancel',
		'id': 'Id',
		'is_available': 'IsAvailable',
		'start_date_time': 'StartDateTime',
		'end_date_time': 'EndDateTime',
		'last_modified_date_time': 'LastModifiedDateTime',
		'class_description': 'ClassDescription',
		'staff': 'Staff',
		'booking_window': 'BookingWindow',
		'booking_status': 'BookingStatus',
		}

	class_schedule_id = None
	visits = None
	clients = None
	location = None
	resource = None
	max_capacity = None
	web_capacity = None
	total_booked = None
	total_booked_waitlist = None
	web_booked = None
	semester_id = None
	is_canceled = None
	substitute = None
	active = None
	is_waitlist_available = None
	is_enrolled = None
	hide_cancel = None
	id = None
	is_available = None
	start_date_time = None
	end_date_time = None
	last_modified_date_time = None
	class_description = None
	staff = None
	booking_window = None
	booking_status = None

File: swagger/Models/Class.php
<?php

namespace App\Mindbody\Models;

/**
 * Represents a single class instance. The class meets at the start time,
 * goes until the end time.
 */
class Class extends BaseModel
{
	/**
	 * The ID used to retrieve the class schedule for the desired class.
	 * 
	 * @var integer format:int32
	 */
	public $classScheduleId;
	/**
	 * Contains information about visits.
	 * 
	 * @var Visit[] 
	 */
	public $visits;
	/**
	 * Contains information about clients.
	 * 
	 * @var Client[] 
	 */
	public $clients;
	/**
	 * Contains information about a location.
	 * 
	 * @var Location 
	 */
	public $location;
	/**
	 * Contains information about a resource, such as a room, assigned to a
	 * class.
	 * 
	 * @var Resource 
	 */
	public $resource;
	/**
	 * The maximum number of clients allowed in the class.
	 * 
	 * @var integer format:int32
	 */
	public $maxCapacity;
	/**
	 * The maximum number of clients allowed to sign up online for the class.
	 * 
	 * @var integer format:int32
	 */
	public $webCapacity;
	/**
	 * The total number of clients booked in the class.
	 * 
	 * @var integer format:int32
	 */
	public $totalBooked;
	/**
	 * The total number of booked clients on the waiting list for the class.
	 * 
	 * @var integer format:int32
	 */
	public $totalBookedWaitlist;
	/**
	 * The total number of clients who signed up online for the class.
	 * 
	 * @var integer format:int32
	 */
	public $webBooked;
	/**
	 * The ID of the semester that the class is a part of, if any.
	 * 
	 * @var integer format:int32
	 */
	public $semesterId;
	/**
	 * When `true`, indicates that the class has been cancelled.<br />  When
	 * `false`, indicates that the class has not been cancelled.
	 * 
	 * @var boolean 
	 */
	public $isCanceled;
	/**
	 * When `true`, indicates that the class is being taught by a substitute
	 * teacher.<br />  When `false`, indicates that the class is being taught
	 * by its regular teacher.
	 * 
	 * @var boolean 
	 */
	public $substitute;
	/**
	 * When `true`, indicates that the class is shown to clients when in
	 * consumer mode.<br />  When `false`, indicates that the class is not
	 * shown to clients when in consumer mode.
	 * 
	 * @var boolean 
	 */
	public $active;
	/**
	 * When `true`, indicates that the clients can be placed on a waiting
	 * list for the class.<br />  When `false`, indicates that the clients
	 * cannot be placed on a waiting list for the class.
	 * 
	 * @var boolean 
	 */
	public $isWaitlistAvailable;
	/**
	 * When `true`, indicates that the client with the given `ClientId` is
	 * enrolled in this class.<br />  When `false`, indicates that the client
	 * with the given `ClientId` is not enrolled in this class.
	 * 
	 * @var boolean 
	 */
	public $isEnrolled;
	/**
	 * When `true`, indicates that this class is hidden when cancelled.<br />
	 * When `false`, indicates that this class is not hidden when cancelled.
	 * 
	 * @var boolean 
	 */
	public $hideCancel;
	/**
	 * The unique identifier for the class.
	 * 
	 * @var integer format:int32
	 */
	public $id;
	/**
	 * When `true`, indicates that the client with the given client ID can
	 * book this class.<br />  When `false`, indicates that the client with
	 * the given client ID cannot book this class.
	 * 
	 * @var boolean 
	 */
	public $isAvailable;
	/**
	 * The time this class is scheduled to start.
	 * 
	 * @var string format:date-time
	 */
	public $startDateTime;
	/**
	 * The time this class is scheduled to end.
	 * 
	 * @var string format:date-time
	 */
	public $endDateTime;
	/**
	 * The last time this class was modified.
	 * 
	 * @var string format:date-time
	 */
	public $lastModifiedDateTime;
	/**
	 * Contains information that defines the class.
	 * 
	 * @var ClassDescription 
	 */
	public $classDescription;
	/**
	 * Contains information about the staff member.
	 * 
	 * @var Staff 
	 */
	public $staff;
	/**
	 * Contains information about the window for booking.
	 * 
	 * @var BookingWindow 
	 */
	public $bookingWindow;
	/**
	 * Contains the booking’s payment status.
	 * 
	 * @var string 
	 */
	public $bookingStatus;

	protected function getInputMap()
	{
		return [
			'ClassScheduleId' => ['classScheduleId', 'integer', 'int32'],
			'Visits' => ['visits', 'array', Visit::class],
			'Clients' => ['clients', 'array', Client::class],
			'Location' => ['location', Location::class, null],
			'Resource' => ['resource', Resource::class, null],
			'MaxCapacity' => ['maxCapacity', 'integer', 'int32'],
			'WebCapacity' => ['webCapacity', 'integer', 'int32'],
			'TotalBooked' => ['totalBooked', 'integer', 'int32'],
			'TotalBookedWaitlist' => ['totalBookedWaitlist', 'integer', 'int32'],
			'WebBooked' => ['webBooked', 'integer', 'int32'],
			'SemesterId' => ['semesterId', 'integer', 'int32'],
			'IsCanceled' => ['isCanceled', 'boolean', null],
			'Substitute' => ['substitute', 'boolean', null],
			'Active' => ['active', 'boolean', null],
			'IsWaitlistAvailable' => ['isWaitlistAvailable', 'boolean', null],
			'IsEnrolled' => ['isEnrolled', 'boolean', null],
			'HideCancel' => ['hideCancel', 'boolean', null],
			'Id' => ['id', 'integer', 'int32'],
			'IsAvailable' => ['isAvailable', 'boolean', null],
			'StartDateTime' => ['startDateTime', 'string', 'date-time'],
			'EndDateTime' => ['endDateTime', 'string', 'date-time'],
			'LastModifiedDateTime' => ['lastModifiedDateTime', 'string', 'date-time'],
			'ClassDescription' => ['classDescription', ClassDescription::class, null],
			'Staff' => ['staff', Staff::class, null],
			'BookingWindow' => ['bookingWindow', BookingWindow::class, null],
			'BookingStatus' => ['bookingStatus', 'string', null],
			];
	}

	protected function getOutputMap()
	{
		return [
			'classScheduleId' => ['ClassScheduleId', 'integer', 'int32'],
			'visits' => ['Visits', 'array', Visit::class],
			'clients' => ['Clients', 'array', Client::class],
			'location' => ['Location', Location::class, null],
			'resource' => ['Resource', Resource::class, null],
			'maxCapacity' => ['MaxCapacity', 'integer', 'int32'],
			'webCapacity' => ['WebCapacity', 'integer', 'int32'],
			'totalBooked' => ['TotalBooked', 'integer', 'int32'],
			'totalBookedWaitlist' => ['TotalBookedWaitlist', 'integer', 'int32'],
			'webBooked' => ['WebBooked', 'integer', 'int32'],
			'semesterId' => ['SemesterId', 'integer', 'int32'],
			'isCanceled' => ['IsCanceled', 'boolean', null],
			'substitute' => ['Substitute', 'boolean', null],
			'active' => ['Active', 'boolean', null],
			'isWaitlistAvailable' => ['IsWaitlistAvailable', 'boolean', null],
			'isEnrolled' => ['IsEnrolled', 'boolean', null],
			'hideCancel' => ['HideCancel', 'boolean', null],
			'id' => ['Id', 'integer', 'int32'],
			'isAvailable' => ['IsAvailable', 'boolean', null],
			'startDateTime' => ['StartDateTime', 'string', 'date-time'],
			'endDateTime' => ['EndDateTime', 'string', 'date-time'],
			'lastModifiedDateTime' => ['LastModifiedDateTime', 'string', 'date-time'],
			'classDescription' => ['ClassDescription', ClassDescription::class, null],
			'staff' => ['Staff', Staff::class, null],
			'bookingWindow' => ['BookingWindow', BookingWindow::class, null],
			'bookingStatus' => ['BookingStatus', 'string', null],
			];
	}
}
